modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequest.java

```diff
@@ -210,28 +155,19 @@ public HttpVersion protocolVersion() {
 
     @Override
     public HttpRequest removeHeader(String header) {
-        HttpHeaders headersWithoutContentTypeHeader = new DefaultHttpHeaders();
-        headersWithoutContentTypeHeader.add(request.headers());
-        headersWithoutContentTypeHeader.remove(header);
-        HttpHeaders trailingHeaders = new DefaultHttpHeaders();
-        trailingHeaders.add(request.trailingHeaders());
-        trailingHeaders.remove(header);
+        HttpHeaders copiedHeadersWithout = request.headers().copy();
+        copiedHeadersWithout.remove(header);
+        HttpHeaders copiedTrailingHeadersWithout = request.trailingHeaders().copy();
+        copiedTrailingHeadersWithout.remove(header);
         FullHttpRequest requestWithoutHeader = new DefaultFullHttpRequest(
             request.protocolVersion(),
             request.method(),
             request.uri(),
             request.content(),
-            headersWithoutContentTypeHeader,
-            trailingHeaders
-        );
-        return new Netty4HttpRequest(
-            sequence,
-            requestWithoutHeader,
-            new HttpHeadersMap(requestWithoutHeader.headers()),
-            released,
-            pooled,
-            content
+            copiedHeadersWithout,
+            copiedTrailingHeadersWithout
         );
+        return new Netty4HttpRequest(sequence, requestWithoutHeader, released, pooled, content);
```

Comment:
Ok, maybe this change here in `Netty4HttpRequest` is mildly interesting...

It's not a big deal in practice because `removeHeader` is only invoked in a specific error condition that's not important wrt to validation/authentication.
But, conceptually, the idea is that the `Netty4HttpRequest` instance, through it's encapsuling `io.netty.handler.codec.http.FullHttpRequest` instance, should have a clearly defined behavior wrt to the validation status, which tags along in yet another encapsulation, i.e. the  `HttpHeaders` instance of the `FullHttpRequest` request.
I've chosen to preserve the validation when removing the headers and unit tested it in `HttpHeadersValidatorTests`. We could chose differently, we can discuss, but it doesn't make any difference in practice because when `removeHeaders` is called the request is already rejected as malformed, and the validation status is not important.